readthedocs/oauth/models.py

```diff
@@ -212,3 +222,36 @@ def matches(self, user):
                 },
             ),
         } for project in projects]
+
+
+class RemoteRepositoryRelation(TimeStampedModel):
+    remote_repository = models.ForeignKey(
+        RemoteRepository,
+        related_name='remote_repository_relations',
+        on_delete=models.CASCADE
+    )
+    user = models.ForeignKey(
+        User,
+        related_name='remote_repository_relations',
+        on_delete=models.CASCADE
+    )
+    account = models.ForeignKey(
+        SocialAccount,
+        verbose_name=_('Connected account'),
+        related_name='remote_repository_relations',
+        on_delete=models.CASCADE
+    )
+    admin = models.BooleanField(_('Has admin privilege'), default=False)
```

Comment:
I'd say that this is "not strictly required". However, for some actions in the old code we tried first with this account (the one the originally imported the repo) and if fail it continues with the following. This reduced the probabilities to fail due lack of permissions. (*)

In case we want to do this here, we will need to modify some code to keep this synced/updated. Besides, if we rely on `imported` for specific tasks we will need to write fallback code in case the original user deletes his account: we won't have the `RemoteRepositoryRelation.imported=True` anymore for that `Project`.

(*) we could use your idea but rename it to `prefered=True` (or similar) that we can set with the first account that works, and update with the following in the list when it fails. Makes sense?

Reply:
I don't think it's important to have then, unless we have a strong reason to have it. Seems like other prioritization methods (eg. owners then team members, etc.) would work fine, but also just "everyone who has access" is fine for now.